src/brevitas_examples/imagenet_classification/ptq/learned_round_utils.py

```diff
@@ -0,0 +1,218 @@
+        self.decay_type = decay_type
+        self.decay_shape = decay_shape
+        self.decay_start = rel_decay_start * t_max
+
+    def __call__(self, t):
+        if t < self.decay_start:
+            return self.start_b
+
+        rel_t = (t - self.decay_start) / (self.t_max - self.decay_start)
+        if self.decay_type == 'linear':
+            return self.end_b + (self.start_b - self.end_b) * max(0.0, (1 - rel_t))
+        elif self.decay_type == 'cosine':
+            return self.end_b + 0.5 * (self.start_b - self.end_b) * (1 + np.cos(rel_t * np.pi))
+        elif self.decay_type == 'sigmoid':
+            d = self.decay_shape
+            offset = sigmoid(-d / 2)
+            rel_progress = (sigmoid(d * (rel_t - 0.5)) - offset) / (1 - 2 * offset)
+            return self.start_b + (self.end_b - self.start_b) * rel_progress
+        elif self.decay_type == 'power':
+            return self.end_b + (self.start_b - self.end_b) * (1 - rel_t ** self.decay_shape)
+        elif self.decay_type == 'exp':
+            r = self.decay_shape
+            rel_progress = (1.0 - np.exp(-r * rel_t)) / (1.0 - np.exp(-r))
+            return self.start_b + (self.end_b - self.start_b) * rel_progress
+        elif self.decay_type == 'log':
+            r = self.decay_shape
+            C = np.exp(self.end_b / r)
+            c = np.exp(self.start_b / r)
+            return r * np.log((C - c) * rel_t + c)
+        else:
+            raise ValueError(f'Unknown temp decay type {self.decay_type}')
+
+
+class CombinedLoss:
+
+    def __init__(
+            self,
+            module,
+            weight=0.01,
+            max_count=1000,
+            b_range=(20, 2),
+            warmup=0.2,
+            decay_start=0.0,
+            **temp_decay_kw):
+        self.weight = weight
+        self.module = module
+        self.loss_start = max_count * warmup
+        self.temp_decay = TempDecay(
+            max_count,
+            b_range=b_range,
+            rel_decay_start=warmup + (1.0 - warmup) * decay_start,
+            **temp_decay_kw,
+        )
+        self.iter = 0
+
+    def __call__(self, pred, tgt, *args, **kwargs):
+        self.iter += 1
+
+        rec_loss = F.mse_loss(pred, tgt, reduction='none').sum(1).mean()
+
+        if self.iter < self.loss_start:
+            b = self.temp_decay(self.iter)
+            round_loss = 0
+        else:  # 1 - |(h-0.5)*2|**b
+            b = self.temp_decay(self.iter)
+            round_vals = self.module.weight_quant.tensor_quant.int_quant.float_to_int_impl.p_forward(
+            )
+            round_loss = self.weight * (1 - ((round_vals - 0.5).abs() * 2).pow(b)).sum()
+
+        total_loss = rec_loss + round_loss
+        return total_loss, rec_loss, round_loss
+
+
+def insert_learned_round_quantizer(layer, learned_round_zeta=1.1, learned_round_gamma=-0.1):
+    if isinstance(layer, QuantWBIOL):
+        if not isinstance(layer.weight_quant.tensor_quant.int_quant.float_to_int_impl,
```

Comment:
Look up this module by instance instead of getting it through this hardcoded path